src/sentry/middleware/staff.py

```diff
@@ -0,0 +1,43 @@
+from django.conf import settings
+from django.utils.deprecation import MiddlewareMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.auth.staff import Staff, logger
+
+
+class StaffMiddleware(MiddlewareMixin):
```

Comment:
`MiddlewareMixin` is deprecated and we should be using the current-style middleware that doesn't need to implement that mixin: https://docs.djangoproject.com/en/5.0/topics/http/middleware/#writing-your-own-middleware